src/fu-util.c

```diff
@@ -923,23 +923,35 @@ fu_util_device_test_remove_emulated_devices(FuUtilPrivate *priv, GError **error)
 	return TRUE;
 }
 
+static gchar *
+fu_util_maybe_expand_basename(const gchar *maybe_basename)
+{
+	if (g_str_has_prefix(maybe_basename, "https://"))
+		return g_strdup(maybe_basename);
+	if (g_str_has_prefix(maybe_basename, "/"))
+		return g_strdup(maybe_basename);
+	return g_strdup_printf("https://fwupd.org/downloads/%s", maybe_basename);
```

Comment:
I don't like having URLs hardcoded. Could we instead discover from fwupd remote?

Reply:
How about using `FirmwareBaseURI`?

Reply:
> How about using FirmwareBaseURI?

If you're running a fwupd mirror you'll be setting that to something custom -- which works for the firmware files, but not for the emulation data. Maybe the fix is "_include FirmwareBaseURI in the shipped lvfs.conf_" and also "_include the emulation data in the PULP_MANIFEST_"

Reply:
Narrator: He did half of that.